agent/reconciler/host_reconciler.go

```
@@ -156,6 +169,29 @@ func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastru
 	return ctrl.Result{}, nil
 }
 
+func (r *HostReconciler) installerController(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) (ctrl.Result, error) {
+	logger := ctrl.LoggerFrom(ctx)
+	installScript, uninstallScript, err := r.getInstallationScript(ctx, byoHost.Spec.InstallationSecret.Name, byoHost.Spec.InstallationSecret.Namespace)
+	if err != nil {
+		logger.Error(err, "error getting installation script")
+		r.Recorder.Eventf(byoHost, corev1.EventTypeWarning, "ReadInstallationSecretFailed", "installation secret %s not found", byoHost.Spec.InstallationSecret.Name)
+		return ctrl.Result{}, err
```

Why return err instead of nil?

---

When `err != nil` we should return the error right?